app/pages/project/classify.cjsx

```diff
@@ -135,14 +164,23 @@ module.exports = React.createClass
     if Math.abs(idealScrollY - scrollY) > lineHeight
       animatedScrollTo document.body, el.offsetTop - space, 333
 
-  handleCompletion: ->
+  saveClassification: ->
     console?.info 'Completed classification', @state.classification
     @state.classification.save().then (classification) =>
       console?.log 'Saved classification', classification.id
+      # After saving, remove the classification resource from the local cache.
       classification.destroy()
 
   loadAnotherSubject: ->
     @getCurrentWorkflowID(@props).then (workflowID) =>
+      # Forget the old classification so a new one will load.
       currentClassifications.forWorkflow[workflowID] = null
-      currentWorkflowForProject[@props.project.id] = null
-      @loadClassification()
+      # Forget the old workflow, unless it was specified, so we'll get a random one next time.
+      unless @props.query.workflow
```

Comment:
Maybe `@props.query?.workflow` here?
I saw a `TypeError: _this.props.query is null` console warning on my initial classification from L179